libs/sdk-core/src/boltzswap.rs

```diff
@@ -85,58 +84,169 @@ struct Pairs {
 impl ReverseSwapperAPI for BoltzApi {
-    async fn reverse_swap_info(&self) -> Result<ReverseSwapPairInfo> {
-        crate::boltzswap::reverse_swap_info().await
+    async fn fetch_reverse_swap_fees(&self) -> Result<ReverseSwapPairInfo> {
+        reverse_swap_pair_info().await
     }
 
-    async fn create_reverse_swap(
+    /// Call Boltz API and parse response as per https://docs.boltz.exchange/en/latest/api/#creating-reverse-swaps
+    ///
+    /// #### Errors
+    ///
+    /// This method returns an error for  HTTP or connection errors (404 not found, 400 bad request,
+    /// 502 server error, etc).
+    ///
+    /// Boltz API errors (e.g. if the reverse swap could not be created, for example if the amount is too low)
+    /// are returned as a successful response of type [BoltzApiCreateReverseSwapResponse::BoltzApiError]
+    async fn create_reverse_swap_on_remote(
         &self,
         amount_sat: u64,
-        onchain_claim_address: String,
+        preimage_hash_hex: String,
+        claim_pubkey: String,
         pair_hash: String,
         routing_node: String,
-    ) -> Result<ReverseSwap> {
-        let rand_bytes: [u8; 32] = random();
-        let preimage = sha256::Hash::hash(&rand_bytes);
-        let preimage_hash = sha256::Hash::hash(&preimage);
-        let preimage_hash_hex = preimage_hash.to_hex();
-
-        let response: CreateReverseSwapResponse = Client::new()
+    ) -> Result<BoltzApiCreateReverseSwapResponse> {
+        let response = Client::new()
             .post(CREATE_REVERSE_SWAP_ENDPOINT)
             .header(CONTENT_TYPE, "application/json")
             .body(get_boltz_reverse_swap_args(
                 amount_sat,
                 preimage_hash_hex,
-                pair_hash,
-                onchain_claim_address,
+                pair_hash.clone(),
+                claim_pubkey.clone(),
                 routing_node,
             ))
             .send()
             .await?
-            .json()
+            .text()
             .await?;
 
-        // TODO In case of error, return Err(str) or Ok(ReverseSwap{ error_message = ..} ) ?
-        return Ok(ReverseSwap {
-            error_message: None,
-            response,
-        });
+        debug!(
+            "Boltz API create response: {}",
+            serde_json::to_string_pretty(&response)?
+        );
+        serde_json::from_str(&response).map_err(|err| anyhow!(err))
+    }
+
+    /// Call and parse response as per https://docs.boltz.exchange/en/latest/api/#getting-status-of-a-swap
+    ///
+    /// #### Errors
+    ///
+    /// This method returns an error for  HTTP or connection errors (404 not found, 400 bad request,
+    /// 502 server error, etc).
+    ///
+    /// Boltz API errors (e.g. providing an invalid ID arg) are returned as a successful response of
+    /// type [BoltzApiCreateReverseSwapResponse::BoltzApiError]
+    async fn get_dynamic_boltz_status(&self, id: String) -> Result<BoltzApiReverseSwapStatus> {
```

Comment:
Why not get_boltz_status?

Reply:
Done in 7dacf9abcd3a63a49ae961fae1a2d02647c2d491